ibis/expr/tests/test_value_exprs.py

```diff
@@ -1340,6 +1340,20 @@ def test_column_isin_map_keys():
     assert isinstance(expr, ir.BooleanColumn)
 
 
+def test_map_get_with_compatible_value_int8():
+    t = ibis.literal({'A': 1000, 'B': 2000})
+    v = t.get('C', 3)
+    assert isinstance(v, ir.IntegerScalar)
+    assert v.type() == dt.int16
+
+
+def test_map_get_with_compatible_value_float():
+    t = ibis.literal({'A': 1000, 'B': 2000})
+    v = t.get('C', 3.0)
```

Comment:
Hm, so this doesn't seem correct to me actually. Computing the precedence is too generic. We only want to be able to upcast within the same numeric kind not across types. non numeric types must be exactly equal.

Reply:
See my comment below.